File: common/src/test/java/org/sqlbuilder/common/TestVariables.java
package org.sqlbuilder.common;

import org.junit.BeforeClass;
import org.junit.Test;

import java.util.ResourceBundle;

public class TestVariables
{
	private static final String[] dollarVs = new String[]{"${var.a}", "${var.b}", "${var.c}", "__${var.a}__", "__${var.b}__", "__${var.c}__", "${var.z}"};
	private static final String[] atVs = new String[]{"@{var.a}", "@{var.b}", "@{var.c}", "__@{var.a}__", "__@{var.b}__", "__@{var.c}__", "@{var.z}"};

	private static Variables variables;

	@BeforeClass
	public static void init()
	{
		ResourceBundle bundle = ResourceBundle.getBundle("Names");
		variables = new Variables(bundle);
	}

	@Test
	public void testVariablesWithDollar()
	{
		testVariables(dollarVs);
	}

	@Test
	public void testVariablesWithAt()
	{
		testVariables(atVs);
	}

	public void testVariables(String[] vs)
	{
		for (var v : vs)
		{
			try
			{
				var w = variables.varSubstitution(v, true);
				System.out.println(w);
			}
			catch (IllegalArgumentException iae)
			{
				System.err.println("Not found; " + v);
			}
		}
	}
}

File: common/src/test/java/org/sqlbuilder/common/TestVariables.kt
package org.sqlbuilder.common

import org.junit.BeforeClass
import org.junit.Test
import org.sqlbuilder.common.Variables.Companion.make
import java.util.*

class TestVariables {

    @Test
    fun testVariablesWithDollar() {
        testVariables(dollarVs)
    }

    @Test
    fun testVariablesWithAt() {
        testVariables(atVs)
    }

    fun testVariables(vs: Array<String>) {
        for (v in vs) {
            try {
                val w: String = variables!!.varSubstitution(v, true)
                println(w)
            } catch (_: IllegalArgumentException) {
                System.err.println("Not found: $v")
            }
        }
    }

    companion object {

        private val dollarVs = arrayOf<String>("\${var.a}", "\${var.b}", "\${var.c}", "__\${var.a}__", "__\${var.b}__", "__\${var.c}__", "\${var.z}")
        private val atVs = arrayOf<String>("@{var.a}", "@{var.b}", "@{var.c}", "__@{var.a}__", "__@{var.b}__", "__@{var.c}__", "@{var.z}")

        private var variables: Variables? = null

        @BeforeClass
        @JvmStatic
        fun init() {
            val bundle = ResourceBundle.getBundle("Names")
            variables = make(bundle)
        }
    }
}
